### ISSUE TITLE
Provide webpack progress output

### ISSUE THREAD

**Issue by aaronbeall** [webpack/webpack-dev-middleware#316]
> The Issues page for this repository is not a support forum. **Please ask questions on StackOverflow or the webpack Gitter (https://gitter.im/webpack/webpack). Questions will be closed.** If you proceed with this form, please fill out **_all_** fields, or your issue may be closed as "invalid." Please do not delete this template from a new issue. Please remove this header to acknowledge this message.

* Operating System: MacOS High Sierra
* Node Version: v8.11.3
* NPM Version: 5.6.0
* webpack Version: 4.16.2
* webpack-serve Version: 2.0.2

- [x] This is a **feature** request
- [ ] This is a **bug**

### Request

I recently migrated from Webpack 3 and `webpack-dev-server` to Webpack 4 and `webpack-serve`. An issue I ran into is that `webpack-serve` doesn't provide any progress output. I logged an issue here: https://github.com/webpack-contrib/webpack-serve/issues/253#issuecomment-408421233
I was advised to log an issue against `webpack-dev-middleware` instead.

### For Features; What is the motivation and/or use-case for the feature?

My initial build can take a couple minutes. Seeing progress output is helpful.

**Comment by evilebottnawi** [comment]
https://github.com/webpack-contrib/webpack-serve#community-add-ons (https://github.com/elisherer/webpack-serve-waitpage)

**Comment by aaronbeall** [comment]
Yes I have tried that, that puts progress in a browser page, this request is for progress in the terminal window. [In the linked issue](https://github.com/webpack-contrib/webpack-serve/issues/253) I explain why progress in the serve browser page isn't right in my case.

**Comment by shellscape** [comment]
@aaronbeall if you're interested in this feature, please open a Pull Request for the feature, as I mentioned in the linked issue.

You are working in a specialized environment, and it's not reasonable to expect a package project to do the work on a feature which (up to this point anyhow) has only been vocalized as being useful to you and your environment. If you'd like to see this feature be part of this package, please invest some time in making that a reality.

**Comment by aaronbeall** [comment]
I find it pretty surprising that command line progress is considered a special need! Must say the transition from `webpack-dev-server` to `webpack-serve` has been bumpy.

**Comment by shellscape** [comment]
Bummer. Thankfully that hasn't been the universal experience.

**Comment by ev1stensberg** [comment]
@aaronbeall Andrew has put a lot of effort in trying to make it sublime, but he can’t cover all cases as well as it would be nice with some help now and then 😊

**Comment by aaronbeall** [comment]
I'd be happy to help but I really have no knowledge about the inner workings of webpack or the dev server. Can you point me in the direction where such a feature could be added? It sounded like you weren't interested in such a feature, though?

**Comment by ev1stensberg** [comment]
I think maybe Andrew would be the most qualified man to answer those questions

**Comment by fmagaldea** [comment]
As a workaround, you can use webpack internal plugin 'ProgressPlugin' https://github.com/webpack/docs/wiki/internal-webpack-plugins#progresspluginhandler.
It will provide you a basic cli output.

In your webpack.config.js:
```javascript
const webpack = require('webpack');
...
module.exports = {
  ...
  plugins: [
    ...
    new webpack.ProgressPlugin()
  ]
}
```

**Comment by michael-ciniawsky** [comment]
@aaronbeall Would the potential solution suggested by https://github.com/webpack/webpack-dev-middleware/issues/316#issuecomment-415341965 be sufficient for your use case ?

**Comment by michael-ciniawsky** [comment]
Feel free to reopen or to fill a new issue if that doesn't work out for you
